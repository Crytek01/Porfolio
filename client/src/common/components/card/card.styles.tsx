import styled from 'styled-components';

import { Typography } from '../typography/typography.component';

export const CardContainer = styled.div`
  border-radius: 11px;
  overflow: hidden;

  display: flex;
  flex-direction: column;
  flex: 0 1 24rem;

  width: 24rem;
  min-height: 28rem;

  transition: transform 0.4s ease;

  box-shadow: 0 0rem 1rem rgb(0 0 0 / 3.5%);

  &:hover {
    transform: scale3d(1.03, 1.03, 1.03);
    box-shadow: 0 0.8rem 4.8rem rgb(0 0 0 / 3%);
  }

  &:active {
    box-shadow: 0 0.8rem 4.8rem rgb(0 0 0 / 10%);
  }
`;

export const CardImage = styled.img`
  border-radius: 11px;

  margin-bottom: 1.2rem;

  width: 100%;
  height: 13.6rem;

  background-size: cover;

  opacity: 95%;
`;

export const TextContainer = styled.div`
  padding: 1rem 2rem;
`;

export const CardTitle = styled(Typography).attrs({
  forwardedAs: 'h3',
  size: 'h3',
  weight: 'medium',
  align: 'center',
})``;

export const CardDescription = styled(Typography).attrs({
  size: 'b4',
  align: 'center',
  lineHeight: 1.4,
});
