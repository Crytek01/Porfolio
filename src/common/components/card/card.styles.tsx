import { withProps } from '@common/hocs/with-props.hoc';
import styled from 'styled-components';

import { Image } from '../image/image.component';
import { Typography } from '../typography/typography.component';

export const CardContainer = styled.div`
  border-radius: 6px 6px 3px 3px;

  border: 1px solid #e9e9e9;

  overflow: hidden;

  display: flex;
  flex-direction: column;

  width: 24rem;
  min-height: 28rem;

  transition: all 0.4s;
  backface-visibility: hidden;
  perspective: 1px;
  -webkit-font-smoothing: subpixel-antialiased;

  box-shadow: 0 1px 1rem rgb(0 0 0 / 1%);

  &:hover {
    transform: scale(1.05);
    box-shadow: 0 8px 4.8rem rgb(0 0 0 / 3%);
  }

  &:active {
    box-shadow: 0 0.8rem 4.8rem rgb(0 0 0 / 10%);
  }
`;

export const CardImage = styled(Image)`
  width: 100%;
  height: 13.6rem;

  background-size: cover;

  opacity: 95%;
`;

export const TextContainer = styled.div`
  padding: 1rem 2rem;
`;

export const CardTitle = withProps(Typography, {
  as: 'h3',
  size: 'h3',
  weight: 'medium',
  align: 'center',
  color: 'lightGrey',
});

export const CardDescription = withProps(Typography, {
  size: 'b4',
  align: 'center',
  lineHeight: 1.4,
});
