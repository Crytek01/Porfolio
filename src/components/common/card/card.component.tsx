import { FunctionComponent } from 'react';
import { IImage } from '../../../contexts/works/works.service';
import {
  CardContainer,
  CardImage,
  CardInformation,
  CardTitle,
  TextContainer,
} from './card.styles';

interface ICard {
  logoImage: IImage;
  description: string;
  title: string;
  information: string;
}

export const Card: FunctionComponent<ICard> = ({
  logoImage,
  description,
  title,
  information,
}) => (
  <CardContainer>
    <picture>
      <source srcSet={logoImage.basic.source} type={logoImage.basic.type} />
      <source
        srcSet={logoImage.optimized.source}
        type={logoImage.optimized.type}
      />
      <CardImage alt={description} src={logoImage.basic.source} />
    </picture>

    <TextContainer>
      <CardTitle>{title}</CardTitle>

      <CardInformation>{information}</CardInformation>
    </TextContainer>
  </CardContainer>
);
