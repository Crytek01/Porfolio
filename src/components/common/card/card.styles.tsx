import styled from 'styled-components';

export const CardContainer = styled.div`
  border-radius: 11px;
  overflow: hidden;

  display: flex;
  flex-direction: column;
  flex: 0 1 24rem;

  width: 24rem;
  height: 30rem;

  //Animations
  transition: transform 0.4s ease;

  &:hover {
    transform: scale3d(1.03, 1.03, 1.03);
    box-shadow: 0 0.8rem 4.8rem rgb(0 0 0 / 3%);
  }
`;

export const CardImage = styled.img`
  border-radius: 11px;

  margin-bottom: 1.2rem;

  width: 100%;
  height: 13.6rem;

  background-size: cover;
`;

export const TextContainer = styled.div`
  padding: 1rem;
`;

export const CardTitle = styled.p`
  font-weight: 500;
  font-size: 2.4rem;
  margin-bottom: 1rem;

  text-align: center;
`;

export const CardInformation = styled.p`
  font-weight: 400;
  font-size: 1.4rem;
  line-height: 1.8;

  color: #555;

  text-align: center;
`;
