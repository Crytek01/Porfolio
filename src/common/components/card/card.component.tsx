import { FunctionComponent } from 'react';

import { IImage } from '../../../contexts/works/works.service';
import { Spacer } from '../spacer/spacer.component';
import { Typography } from '../typography/typography.component';

import { CardContainer, CardImage, TextContainer } from './card.styles';

interface ICard {
  logoImage: IImage;
  description: string;
  title: string;
  information: string;
}

export const Card: FunctionComponent<ICard> = ({
  logoImage,
  description,
  title,
  information,
}) => (
  <CardContainer>
    <CardImage
      alt={description}
      src={logoImage.basic.source}
      type={logoImage.basic.type}
      optimizedSrc={logoImage.optimized.source}
      optimizedType={logoImage.optimized.type}
    />

    <Spacer size={3} />

    <TextContainer>
      <Typography
        as="h3"
        size="h3"
        weight="medium"
        align="center"
        color="semiBoldGrey"
      >
        {title}
      </Typography>

      <Spacer size={2} />

      <Typography size="b4" align="center" lineHeight={1.4} color="lightGrey">
        {information}
      </Typography>
    </TextContainer>
  </CardContainer>
);
