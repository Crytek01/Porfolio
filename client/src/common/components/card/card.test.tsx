import renderer from 'react-test-renderer';
import { describe, expect, it } from 'vitest';

import { Card } from './card.component';

describe('Card component Test', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Card
          description="description"
          information="information"
          logoImage={{
            basic: { type: 'png', source: 'basic image' },
            optimized: { type: 'webp', source: 'optimized image' },
          }}
          title="title"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
