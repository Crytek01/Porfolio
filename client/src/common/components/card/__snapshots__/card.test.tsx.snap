// Vitest Snapshot v1

exports[`Card component Test > renders correctly 1`] = `
<div
  className="sc-gKPRtg eA-dsLn"
>
  <div
    className="sc-hLBbgP gAvAzt sc-iBYQkv ftFZOW"
  >
    <picture>
      <source
        srcSet="optimized image"
        type="webp"
      />
      <source
        srcSet="basic image"
        type="png"
      />
      <img
        alt="description"
        className="sc-eDvSVe gxsJM sc-iBYQkv ftFZOW"
        onLoad={[Function]}
        src="basic image"
      />
      <div
        className="sc-dkrFOg jGJVgg sc-jSUZER gHhgjy"
        size={50}
      />
    </picture>
  </div>
  <div
    className="sc-ftTHYK dGFSPJ"
  >
    <h3
      className="sc-gswNZR gXqiCu"
    >
      title
    </h3>
    <div
      className="sc-bcXHqe kiYcIC"
      size={2}
    />
    <p
      className="sc-gswNZR dgIsqM"
    >
      information
    </p>
  </div>
</div>
`;
